### PR TITLE
[Transactionn] Transaction admin api getTransactionBufferStatus and getPendingAckStatus

### PR DESCRIPTION
**Pull request by congbobo184**
## Motivation
Transaction add admin api `getTransactionBufferStatus`
Transaction add admin api `getPendingAckStatus`

## implement
```
    @Data
    public static class TransactionBufferStatus {
        /** The state of this transaction buffer. */
        public String state;

        /** The max read position of this transaction buffer. */
        public String maxReadPosition;

        /** The last snapshot timestamps of this transaction buffer. */
        public long lastSnapshotTimestamps;
    }

    @Data
    public static class TransactionPendingAckStatus{
        /** The state of this pending ack. */
        public String state;
    }
```
This is transaction component status.
### Verifying this change
Add the tests for it

Does this pull request potentially affect one of the following parts:
If yes was chosen, please highlight the changes

Dependencies (does it add or upgrade a dependency): (no)
The public API: (no)
The schema: (no)
The default values of configurations: (no)
The wire protocol: (no)
The rest endpoints: (no)
The admin cli options: (yes)
Anything that affects deployment: (no)

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java [hunk 1]
```diff
@@ -55,6 +55,55 @@ public void getCoordinatorStatus(@Suspended final AsyncResponse asyncResponse,
...
     }
 
+    @GET
+    @Path("/transactionInBufferStats")
+    @ApiOperation(value = "Get transaction state in transaction buffer.")
+    @ApiResponses(value = {@ApiResponse(code = 403, message = "Don't have admin permission"),
+            @ApiResponse(code = 404, message = "Tenant or cluster or namespace or topic doesn't exist"),
+            @ApiResponse(code = 503, message = "This Broker is not configured "
+                    + "with transactionCoordinatorEnabled=true."),
+            @ApiResponse(code = 307, message = "Topic don't owner by this broker!"),
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java]: what about
"Topic is not owned by this broker"

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java [hunk 2]
```diff
@@ -55,6 +55,55 @@ public void getCoordinatorStatus(@Suspended final AsyncResponse asyncResponse,
...
+            @ApiResponse(code = 501, message = "Topic is not a persistent topic!"),
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java]: "Topic is not a persistent topic"

501 is kind of "internal server error", probably we should return something like "Not allowed" or "Bad request"

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java [hunk 3]
```diff
@@ -55,6 +55,55 @@ public void getCoordinatorStatus(@Suspended final AsyncResponse asyncResponse,
...
+            @ApiResponse(code = 409, message = "Concurrent modification")})
+    public void getTransactionInBufferStats(@Suspended final AsyncResponse asyncResponse,
+                                            @QueryParam("authoritative")
+                                            @DefaultValue("false") boolean authoritative,
+                                            @QueryParam("mostSigBits")
```
Comment (eolivelli) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java]: isn't it a little bit awkward to  to pass separately mostSigBits and leastSigBits in a REST API ?
can we using a string representation ?

do we have other examples in the REST API about splitting the TxID into the two parts ?

**Review by eolivelli — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java (hunk 3 above)
Comment (congbobo184) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java]: client don't know the txnId is consists of coordinatorId and sequenceId. it only know a TxnID consists of mostSigBits and leastSigBits. It's better to understand.

**Review by codelipenghui — CHANGES_REQUESTED** [review]
Please check all the name which use `**Status`, we are using `**Stats` before.

**Comment by congbobo184** [comment]
@Anonymitaet please review doc.

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Transactions.java [hunk 4]
```diff
@@ -71,4 +73,21 @@
...
+     * @param topic the topic of getting transaction buffer stats
+     * @return the future stats of transaction buffer in topic.
+     */
+    CompletableFuture<TransactionBufferStats> getTransactionBufferStats(String topic);
+
+    /**
+     * Get transaction pending ack stats.
+     *
+     * @param topic the topic of this transaction pending ack stats
+     * @param subName the topic of this transaction pending ack stats
```
Comment (Anonymitaet) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/client/admin/Transactions.java:88]:
```suggestion
     * @param subName the subName of this transaction pending ack stats
```
is `subName` accurate? or `subscription name`?

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@congbobo184 Could you please update the title of the PR?

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java (hunk 3 above)
Comment (codelipenghui) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v3/Transactions.java]: @eolivelli Currently all the txn Id in the admin API expressed as `mostSigBits` and `leastSigBits`. The client also can get the `mostSigBits` and `leastSigBits` of a transactionId when using the transactions.

**Review by eolivelli — APPROVED** [review]
perfect

**Pull request merged**